cpp/arcticdb/storage/python_bindings.cpp

```diff
@@ -100,6 +100,13 @@ void register_bindings(py::module& storage, py::exception<arcticdb::ArcticExcept
                                             return none.cast<py::object>();
                                        });
         })
+        .def("check_primary_storage_is_accessible", [](Library& library) {
+            std::optional<bool> out;
+            library.storage_specific([&out](storage::s3::S3Storage& s3) {
+                out = s3.check_creds_and_bucket();
+            });
+            return out;
+        }, "Currently only implemented by S3 storage. Calling on other storage types does nothing.")
```

Comment:
I think we should document what it does as well as what it's implemented for. Something like:

`For S3 storages will _HEAD_ the bucket to test connectivity. Storage types other than S3 are not implemented - calling this will return _None_`.

Also, should we raise a `NotImplementedException` or something?